cocos/gfx/webgpu/webgpu-descriptor-set.ts

```diff
@@ -228,70 +224,20 @@ export class WebGPUDescriptorSet extends DescriptorSet {
         return false;
     }
 
-    private _isResourceChange (binds: number[]): boolean {
-        return binds.every((bind) => {
-            const resource = this._buffers[bind] as WebGPUBuffer || this._textures[bind] || this._samplers[bind];
-            return !this._hasResourceChange(bind, resource);
+    private _isResourceChange (): boolean {
+        const layout = this._layout as WebGPUDescriptorSetLayout;
+        return layout.gpuDescriptorSetLayout!.bindings.every((bind) => {
+            const binding = bind.binding;
+            const resource = this._buffers[binding] as WebGPUBuffer || this._textures[binding] || this._samplers[binding];
+            return !this._hasResourceChange(binding, resource);
         });
     }
 
-    public prepare (frequency: DescUpdateFrequency, binds: number[], vertBinds: number[] = [], fragBinds: number[] = []): void {
-        // set null
-        const layout = this._layout as WebGPUDescriptorSetLayout;
-        if (!binds) {
-            const device = WebGPUDeviceManager.instance;
-            if (!this._bindGroupEntries.size) {
-                const gpuSetLayout = device.defaultResource.setLayout as WebGPUDescriptorSetLayout;
-                const binding = gpuSetLayout.bindings[0];
-                this._bindBufferEntry(binding, device.defaultResource.buffer);
-                this._needUpdate = true;
-                this.prepare(frequency, [-1]);
-            } else {
-                this.prepare(frequency, [-1]);
-            }
-            // this._resetPrepareDynOffsets();
-            return;
-        }
-        const isLowFrequency = frequency === DescUpdateFrequency.LOW;
-        const breakUpdate = !this._needUpdate && (isLowFrequency ? this._gpuDescriptorSet!.bindGroup
-            : (isBound(binds, this._currBinds) && isBound(binds, layout.currBinds)))
-        && this._hasIncludeDynamic() && this._isResourceChange(binds);
+    public prepare (force: boolean = false): void {
+        const breakUpdate = this._isResourceChange() && !force;
```

Comment:
Here some comments might be added. What does `prepare` do, and why `_isResourceChange` is needed can be explained.